src/bartiq/symbolics/sympy_backends.py

```
@@ -275,6 +279,58 @@ def sequence_prod(
         """Express a product of terms expressed using `iterator_symbol`."""
         return sympy.Product(term, (iterator_symbol, start, end))
 
+    @staticmethod
+    def inspect_expression(expression: sympy.Basic) -> None:
+        """Inspect a sympy expression for potential issues.
+
+        This method is useful for investigating an expression that
+        will unexpectedly not evaluate; if no unknown functions are found in
+        the expression this function will not return anything.
+
+        Args:
+            expression (sympy.Basic): The sympy expression to inspect.
+
+        Raises:
+            ValueError: If a operation in the provided expression is not recognised.
+        """
+        ops = _unpack_expression_into_operations(expression=expression)
+
+        known_functions: set[str] = set(_ALL_SYMPY_CONSTANTS + _ALL_SYMPY_OPS + BUILT_IN_FUNCTIONS)
+        potentially_unknown_functions: list[str] = list(ops - known_functions)
+        if potentially_unknown_functions:
+            closest_match: list[str] = difflib.get_close_matches(
+                word=potentially_unknown_functions[0], possibilities=known_functions
+            )
+            msg = f"Unrecognised function call '{potentially_unknown_functions[0]}'."
```

It only acts on the first function.
Consider the following expression:
`g(7) + log_2(10) + 5`.
It will fail on `g(7)`, even though the user might actually not care about it and doesn't want to evaluate.
But it will fail to notify user about the log_2. 
We could return an error message that will just list all the functions? Or provide an option to add "user defined known functions", which are not built in?

---

aha, this is an excellent suggestion. Truthfully I hadn't considered user-defined functions, so only taking the first arg of `potentiall_unknown_functions` seemed reasonable, but allowing the user to provide a list of function names to ignore, I love this idea.